### DISCUSSION TITLE
Difference in enthalpy values for adiabatic compression

### DISCUSSION THREAD

**Discussion by helios13** [CoolProp/CoolProp#2363] [Q&A] [answered]
Hello, I have written a python code for refrigeration cycle using Isopentane as refrigerant. 

Evaporator temperature is T1 (45C), condenser temperature is T2 (100C). Assuming ideal adiabatic compression, I have calculated the work done. The enthalpy at compressor outlet, h2 is equal to enthalpy at compressor inlet, h1 + work done. h1 = CP.PropsSI('H', 'T', T1, 'Q', 1, 'Isopentane'). Other way to calculate enthalpy at state 2 is h2' is CP.PropsSI('H', 'T', T2, 'P', P2, 'Isopentane'), where P2 = P1*(T2/T1)^(y/(y-1)), where y is ratio of Cp and Cv, P1 (176124.18 Pa) is pressure at compressor inlet, P2 is pressure at compressor outlet. Work done = R*(T2-T1)/(y-1)/M, where R is universal gas constant, M is molecular mass in kg/mol. 

My question is why are the two enthalpies (h2 and h2') different? Is it because ideal adiabatic compression doesn't apply to isopentane or am I making some mistake? 

Also when I calculate P2' using CP.PropsSI('P', 'T', T2, 'Q', 1, 'Isopentane'), there is a difference in values from P2 calculated earlier. I expect state 2 to be within the dome, so P2 and P2' should have same values. 

Any help would be appreciated. 
Thanks!

**Comment by ibell** [comment, marked as answer]
The calculation of $P_2$ from isentropic and constant heat capacity assumption for an ideal gas (it is not a good assumption to treat isopentane as an ideal gas) is an entirely different thing from the calculation based on the saturation temperature. The latter method is what you should be using. And then you should calculate the change in specific enthalpy ALWAYS from $h_2-h_1$. And the isentropic outlet state  should be calculated with P,S inputs, with the entropy equal to that at the inlet of the compressor.

**Reply by helios13** [reply]
Thanks. That makes complete sense.
